docs/Frequently Asked Questions/React Native/index.md

```diff
@@ -0,0 +1,247 @@
+        self.sdlManager = notification.userInfo["sdlManager"]
+    }
+
+    sendEvent(withName: "DoAction", body: ["type": "actionType"])
+}
+
+override func supportedEvents() -> [String]! {
+    return ["DoAction"]
+}
+
+}
+```
+
+The above example will call into your JavaScript code with an event type `DoAction`. Inside your React Native (JavaScript) code, create an  `NativeEventEmitter` object within your `EventEmitter` module and add a listener for the event.
+
+```javascript
+import { NativeEventEmitter, NativeModules } from 'react-native';
+const  { SDLEventEmitter } = NativeModules;
+
+const testEventEmitter = new NativeEventEmitter(SDLEventEmitter);
+
+// Build a listener to listen for events
+const testData = testEventEmitter.addListener(
+    'DoAction',
+        () => SDLEventEmitter.eventCall({
+            "data": {
+                "low": "77",
+                "high": "87",
+                "currentTemp": "82",
+                "rain": "50%"
+            }
+        }   
+    )
+)
+```
+### Exposing Methods
+The last step is to wrap the method you wish to expose inside an `RCT_EXPORT_METHOD` for Objective-C and `RCT_EXTERN_METHOD` for Swift. 
+
+Inside the `SDLEventEmitter.m` file make sure to add the following method.
+##### Objective-C
+```objc
+RCT_EXPORT_METHOD(eventCall:(NSDictionary *)dict) {
+    [self.sdlManager.screenManager beginUpdates];
+
+    self.sdlManager.screenManager.textField1 = [NSString stringWithFormat:@"Low: %@ ºF", [RCTConvert NSString:dict[@"data"][@"low"]]];
+    self.sdlManager.screenManager.textField2 = [NSString stringWithFormat:@"High: %@ ºF", [RCTConvert NSString:dict[@"data"][@"high"]]];
+
+    [self.sdlManager.screenManager endUpdatesWithCompletionHandler:^(NSError * _Nullable error) {
+        if (error != nil) {
+            <#Error#>
+        } else {
+            <#Success#>
+        }
+    }];
+}
+```
+##### Swift
+
+!!! NOTE
+Make sure you add `#import "React/RCTEventEmitter.h"` to the apps bridging header.
+!!!
+
+If you're making a React Native application and using native Swift code, you will need to create the Objective-C bridger for the `SDLEventEmitter` class you created above. Wrap the method(s) you wish to expose in a  `RCT_EXTERN_METHOD` macro inside your wrapper class. This wrapper will allow the JavaScript code to talk with your native code.
+
+```objc
+#import "React/RCTBridgeModule.h"
+#import "React/RCTEventEmitter.h"
+@interface RCT_EXTERN_MODULE(SDLEventEmitter, RCTEventEmitter)
+
+RCT_EXTERN_METHOD(eventCall:(eventCall: (id)dict))
+
+@end
+```
+Add the following method to `SDLEventEmitter.swift`:
+
+```swift
+@objc func eventCall(_ dict: NSDictionary) {
+    self.sdlManager.screenManager.beginUpdates()
+    let data = dict["data"]! as! NSDictionary
+    self.sdlManager.screenManager.textField1 = "Low: \(data["low"]!) ºF")"
+    self.sdlManager.screenManager.textField2 = "High: \(data["high"]!) ºF")"
+    self.sdlManager.screenManager.endUpdates()
+}
+```
```

Comment:
You should add some concluding comments about what has been accomplished in the above guide

Reply:
How about now, just updated.